applications/compressible_navier_stokes/taylor_green/application.h

```diff
@@ -296,15 +296,16 @@ class Application : public ApplicationBase<dim, Number>
     pp_data.kinetic_energy_data.calculate                  = true;
     pp_data.kinetic_energy_data.calculate_every_time_steps = 1;
     pp_data.kinetic_energy_data.viscosity                  = DYN_VISCOSITY / RHO_0;
-    pp_data.kinetic_energy_data.filename = this->output_directory + this->output_name;
+    pp_data.kinetic_energy_data.directory                  = this->output_directory;
+    pp_data.kinetic_energy_data.filename                   = this->output_name;
 
     // kinetic energy spectrum
     pp_data.kinetic_energy_spectrum_data.calculate                     = true;
     pp_data.kinetic_energy_spectrum_data.calculate_every_time_steps    = -1;
     pp_data.kinetic_energy_spectrum_data.calculate_every_time_interval = 0.5;
-    pp_data.kinetic_energy_spectrum_data.filename =
-      this->output_directory + this->output_name + "_spectrum";
-    pp_data.kinetic_energy_spectrum_data.degree                     = degree;
+    pp_data.kinetic_energy_spectrum_data.directory                     = this->output_directory;
+    pp_data.kinetic_energy_spectrum_data.filename = this->output_name + "_spectrum";
+    pp_data.kinetic_energy_spectrum_data.degree   = degree;
```

Comment:
This looks like a lot of work. You could have used `std::filesystem::path(file_name).parent_path().to_string();` or something similar to get to the same result.

Reply:
Good to know. On the other hand, I wanted to unify the code in ExaDG anyways and always use parameters for directory + filename (we currently have a mixture of different styles). I think this style is more intuitive for the user.